drivers/console/uart_console.c

```diff
@@ -335,6 +339,114 @@ static void handle_ansi(u8_t byte, char *line)
 	atomic_clear_bit(&esc_state, ESC_ANSI);
 }
 
+#ifdef CONFIG_UART_CONSOLE_MCUMGR
+
+static void clear_mcumgr(void)
+{
+	atomic_clear_bit(&esc_state, ESC_MCUMGR_PKT);
+	atomic_clear_bit(&esc_state, ESC_MCUMGR_PKT_2);
+	atomic_clear_bit(&esc_state, ESC_MCUMGR_DATA);
+	atomic_clear_bit(&esc_state, ESC_MCUMGR_DATA_2);
+}
+
+/**
+ * These states indicate whether an mcumgr frame is being received.
+ */
+#define CONSOLE_MCUMGR_STATE_NONE       1
+#define CONSOLE_MCUMGR_STATE_HEADER     2
+#define CONSOLE_MCUMGR_STATE_PAYLOAD    3
+
+static int read_mcumgr_byte(uint8_t byte)
+{
+	bool data_1;
```

Comment:
The naming scheme is inconsistent;

data_1, _DATA
data_2, _DATA_2

Reply:
I would have preferred to use `data_1` everywhere.  I used `ESC_MCUMGR_DATA` to be consistent with the other enum names.

When I change the variable names to `pkt` and `data`, I think it hurts readability.  I have changed the enums to `ESC_MCUMGR_FRAG_1` and `ESC_MCUMGR_PKT_1`.